### PR TITLE
CATL-1237: Make user menu configurable

### PR DESCRIPTION
**Pull request by reneolivo**
## Overview

This PR allows the user menu introduced in #1 to be configurable so administrators can control showing or hiding this new menu.

## Before
Misc  Undelete  PDFs  Limits  Logging  Captcha  etc     CiviCase
(the option does not exist)

## After
pr-after

## Technical details

In order to add this new field to the miscellaneous settings we followed the steps in https://github.com/compucorp/uk.co.compucorp.civicase/pull/71

The only change of note is the `table.form-layout:eq(1) tr:last` selector. This was chosen because there is no reference to select a specific table where to place the new field so `:eq(1)` was needed instead of a more specific ID or class. Also, we didn't add the field as the last element in the form to avoid mixing it with the ReCAPTCHA settings fields.

### CODE REVIEW

File: usermenu.php [hunk 1]
```diff
@@ -169,6 +169,28 @@ function usermenu_civicrm_navigationMenu(&$menu) {
   _usermenu_civix_navigationMenu($menu);
 } // */
 
+/**
+ * Implements hook_civicrm_alterContent().
+ * Adds extra settings fields to the Civicase Admin Settings form.
+ *
+ * @link https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterContent/
+ */
+function usermenu_civicrm_alterContent (&$content, $context, $templateName, $form) {
```
Comment (tunbola) [inline usermenu.php]: The last parameter is actually an object not necessarily a form object. Could be a page or form object. 
See: https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_alterContent/
We can change to `$object`

File: usermenu.php [hunk 2]
```diff
@@ -169,6 +169,28 @@ function usermenu_civicrm_navigationMenu(&$menu) {
...
+    return;
+  }
+
+  $settingsTemplate = &CRM_Core_Smarty::singleton();
+  $settingsTemplateHtml = $settingsTemplate->fetchWith('CRM/Usermenu/Admin/Form/Settings.tpl', []);
+
+  $doc = phpQuery::newDocumentHTML($content);
+  $doc->find('table.form-layout:eq(1) tr:last')->append($settingsTemplateHtml);
+
+  $content = $doc->getDocument();
```
Comment (tunbola) [inline usermenu.php]: Code is good for now, but if we keep adding new logic in this hook, will be a good time to move the code to separate classes.

File: usermenu.php [hunk 3]
```diff
@@ -178,3 +200,19 @@ function usermenu_civicrm_coreResourceList(&$items, $region) {
...
+  $isViewingMiscSettingsForm = $formName === CRM_Admin_Form_Setting_Miscellaneous::class;
+
+  if (!$isViewingMiscSettingsForm) {
+    return;
+  }
+
+  $settings = $form->getVar('_settings');
+  $settings['allowCivicrmUserMenu'] = CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME;
+
+  $form->setVar('_settings', $settings);
```
Comment (tunbola) [inline usermenu.php]: Code is good for now, but if we keep adding new logic in this hook, will be a good time to move the code to separate classes or we can do it now?

File: usermenu.php (hunk 2 above)
Comment (reneolivo) [reply usermenu.php]: Yes, I know we should use the hook pattern as in other extensions. I'll keep this in mind as soon as we touch this file again.

**Pull request merged**